fiona/_transform.pyx

```diff
@@ -263,3 +264,63 @@ def _transform_geom(src_crs, dst_crs, geom, antimeridian_cutting, antimeridian_o
     OSRRelease(dst)
 
     return out_geom
+
+
+from libcpp.unordered_map cimport unordered_map
+
+cdef OGRGeometryH segmentize(OGRGeometryH geom, double max_length):
+    cdef OGRGeometryH new_geom = NULL
+    new_geom = OGR_G_Clone(geom)
+    OGR_G_Segmentize(new_geom, max_length)
+    return new_geom
+
+
+ctypedef OGRGeometryH (*geometry_func)(OGRGeometryH, double)
+cdef unordered_map[int, geometry_func] func_map
+
+func_map[1] = <geometry_func>segmentize
+
+cdef geometry_func func_alias(int choice):
+    return func_map[choice]
```

Comment:
We can't just put `OGR_G_Segmentize` (for example) in a Python dict. https://stackoverflow.com/a/56708967 was very helpful in showing a solution.